### PR TITLE
`xtask`: Add `r-cmd-check`

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
Prototype implementation based on calling `rcmdcheck::rcmdcheck()`. We can do it manually, but it requires using a temp dir, copying files there, running `R CMD build` and only then `R CMD check`. Let's resort to using a proven R package for now.

Reuses the same Cargo rewriting mechanism as `devtools-test`, which is now improved & stores original Cargo.toml as bytes (no cr/lf issue).

### CODE REVIEW

**Comment by Ilia-Kosenkov** [comment]
I might consider calling this task on CI instead of the weird magic that we do right now.

File: xtask/src/commands/r_cmd_check.rs [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
-    cmd.run()
+
+    let args = format!("c({0})", args.join(", "));
+
+    shell
+        .cmd("Rscript")
+        .arg("-e")
+        .arg(format!(
+            "rcmdcheck::rcmdcheck(args = {}, error_on = 'warning')",
+            args
```
Comment (CGMossa) [inline xtask/src/commands/r_cmd_check.rs:27]:
```suggestion
            "rcmdcheck::rcmdcheck(args = {args}, error_on = 'warning')"
```

**Review by CGMossa — APPROVED** [review]
I don't understand why all the line changes.. did you accidentally change line endings of your files?

We need to just make a `.gitattributes`, and make all line endings conform to the linux once and be done with it once and for all. 

Update: Works!

**Comment by CGMossa** [comment]
Output:
```


❯ checking package dependencies ... ERROR
  Packages suggested but not available: 'patrick', 'lobstr'

  The suggested packages are required for a complete check.
  Checking can be attempted without them by setting the environment
  variable _R_CHECK_FORCE_SUGGESTS_ to a false value.

  See section 'The DESCRIPTION file' in the 'Writing R Extensions'
  manual.

❯ checking CRAN incoming feasibility ... [11s] NOTE
  Maintainer: 'Claus O. Wilke <wilke@austin.utexas.edu>'

  New submission

  Found the following (possibly) invalid URLs:
    URL: https://opensource.org/licenses/MIT (moved to https://opensource.org/license/mit/)
      From: README.md
      Status: 200
      Message: OK
```

**Comment by CGMossa** [comment]
code to install all packages from a `DESCRIPTION` file..

```r
require("attachment")
attachment::install_from_description("tests/extendrtests/DESCRIPTION")
```

**Comment by Ilia-Kosenkov** [comment]
I rolled back CI/CD changes here, I will introduce them separately. It seems `xtask` is working smoothly on CI.

**Review by CGMossa — APPROVED** [review]
LGTM

**Pull request merged**